pkg/segment/writer/metrics/wal/wal.go

```diff
@@ -333,3 +337,202 @@ func (it *WalIterator) decodeWALBlock(blockBuf *bytes.Reader) error {
 
 	return nil
 }
+
+/*
+Write-Ahead Logging (WAL) for Metrics Names
+*/
+
+type MNameWalIterator struct {
+	fd               *os.File
+	readBuf          []byte
+	readMetricsNames []string
+	currentIndex     int
+}
+
+func (it *MNameWalIterator) Close() error {
+	if it.fd != nil {
+		return it.fd.Close()
+	}
+	return errors.New("file descriptor is nil")
+}
+
+/*
+This function appends a new block to the file.
+
+File Format:
+	Version:                1 Byte  // File format version
+
+	// Repeating Blocks Structure
+	[Block] {
+		BlockLen:               4 Bytes  // Size of this block (excluding this field)
+		Checksum:               4 Bytes  // CRC32 checksum for data integrity
+		ZstdEncoded Block:
+			Metrics Names:  	[] 		 // List of metric names stored in the block
+				|-- [Metric Name] {
+					StringLength:  2 Bytes   // Length of the metric name (uint16)
+					StringData:    Variable  // Actual metric name (UTF-8 encoded)
+				}
+	}
+
+	Multiple such blocks are appended continuously.
+*/
+
+func (w *WAL) AppendMName(mName []string) error {
```

Comment:
nit: since this is a list
```suggestion
func (w *WAL) AppendMName(mNames []string) error {
```